images/agent-sandbox-controller/Dockerfile

```diff
@@ -1,21 +1,28 @@
-# Build go binaries
-# See https://github.com/golang/go/issues/69255#issuecomment-2523276831
-FROM --platform=$BUILDPLATFORM golang:1.25.1 AS builder
+ARG GO_VERSION="1.25.3"
 ARG TARGETARCH
 
-COPY go.mod go.sum /workspace/
-COPY api/ /workspace/api/
-COPY cmd/ /workspace/cmd/
-COPY controllers/ /workspace/controllers/
-COPY extensions/ /workspace/extensions/
+# Build go binaries
+# See https://github.com/golang/go/issues/69255#issuecomment-2523276831
+FROM --platform=$BUILDPLATFORM golang:${GO_VERSION} AS builder
 
 WORKDIR /workspace
 
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=${TARGETARCH} go build -o /agent-sandbox-controller ./cmd/agent-sandbox-controller
+# Download dependencies first to leverage layer caching
+COPY go.mod go.sum ./
+RUN go mod download
```

Comment:
If you want to be even faster you can add some prebuilding here!

(e.g. `CGO_ENABLED=0  go build k8s.io/client-go/...`)